megatron/text_generation_utils.py

```diff
@@ -738,7 +743,18 @@ def generate_samples_interactive(
 
         if torch.distributed.is_initialized() and torch.distributed.get_rank() == 0:
             os.system("clear")
-            raw_text = input("Context prompt >>> ")
+            raw_text = ""
+            while True:
+                current_input = input("Context prompt >>> ")
+                if prompt_end == "\n":
```

Comment:
Can you elaborate on why this part of the code treats `\n` as a special case?

Reply:
python's `input` function strips the newline character, so while it reads multiple lines it combines them into a single line input unless it is returned back
I got that behavior when testing, so it's not a speculation

Reply:
Ah I see. I had forgotten about that and assumed you were handling the default case separately for some reason.

Could you add a note about this for future maintainers? Otherwise it looks good and if it tests well I’ll merge it.

Reply:
Added them now in [437bbf5](https://github.com/EleutherAI/gpt-neox/pull/754/commits/437bbf5d42b4df14a43eed4a9db97aebdc67d82d)